pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -490,6 +486,10 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 			Sni: tls.Sni,
 		}
 
+		// Set default SNI of cluster name for istio_mutual if sni is not set.
+		if len(tls.Sni) == 0 && tls.Mode == networking.TLSSettings_ISTIO_MUTUAL {
+			cluster.TlsContext.Sni = cluster.Name
```

Comment:
Doesn't cluster.Name include "|" ? I don't think it's valid in the SNI ( according the the spec it must be a DNS name)

Reply:
Yes, per "spec" it has to be valid DNS name.. But I confirmed that boring SSL doesn't care at all (confirmed with @PiotrSikora as well). Infact, a few months ago, the SNI was set to the cluster name in our code base and things worked normally. 

Given that we set cluster name as SNI only for istio services using mTLS, we know that the other side is Envoy and that envoy will accept arbitrary strings as SNI, we should be safe,